include/geode/geosciences_io/model/helpers/geos_export.h

```diff
@@ -0,0 +1,497 @@
+    class GeosExporterImpl
+    {
+        OPENGEODE_DISABLE_COPY_AND_MOVE( GeosExporterImpl );
+
+    public:
+        GeosExporterImpl(
+            absl::string_view files_directory, const Model& model )
+            : model_( model ),
+              model_curve_{},
+              model_surface_{},
+              model_solid_{},
+              region_attribute_{},
+              files_directory_{ ghc::filesystem::path{
+                  geode::to_string( files_directory ) }
+                                    .string() },
+              prefix_{ geode::filename_without_extension( files_directory ) },
+              cell_1Dproperty_names_{},
+              cell_2Dproperty_names_{},
+              cell_3Dproperty_names_{},
+              well_perforations_{}
+        {
+            std::tie( model_curve_, model_surface_, model_solid_ ) =
+                geode::convert_brep_into_curve_and_surface_and_solid( model_ );
+            static constexpr auto REGION_ID_ATTRIBUTE_NAME{ "attribute" };
+            region_attribute_ =
+                model_solid_->polyhedron_attribute_manager()
+                    .template find_or_create_attribute<
+                        geode::VariableAttribute, geode::index_t >(
+                        REGION_ID_ATTRIBUTE_NAME, geode::NO_ID );
+            if( ghc::filesystem::path{ geode::to_string( files_directory ) }
+                    .is_relative() )
+            {
+                ghc::filesystem::create_directory(
+                    ghc::filesystem::current_path() / files_directory_ );
+            }
+            else
+            {
+                ghc::filesystem::create_directory( files_directory_ );
+            }
+        }
+
+        virtual ~GeosExporterImpl() = default;
+
+        void write_files() const
+        {
+            auto geometry_file = write_well_perforations_boxes();
+            write_well_perforation_file();
+            auto mesh_file = write_mesh_files();
+
+            const auto simu_xml = absl::StrCat(
+                files_directory(), "/", prefix(), "_simulation.xml" );
+            std::ofstream file( simu_xml, std::ofstream::out );
+            file << "<?xml version=\" 1.0 \"?>" << std::endl;
+            file << "<Problem>" << std::endl;
+            file << "<!-- include mesh xml file -->" << std::endl;
+            file << "    <Included>" << std::endl;
+            file << "        <File name = \"./" << mesh_file << "\"/>"
+                 << std::endl;
+            file << "    </Included>" << std::endl;
+            file << "    <!-- include geometry xml file -->" << std::endl;
+            file << "    <Included>" << std::endl;
+            file << "        <File name = \"./" << geometry_file << "\"/>"
+                 << std::endl;
+            file << "    </Included>" << std::endl;
+            file << "    <!-- please complete simulation file here -->"
+                 << std::endl;
+            file << "</Problem>" << std::endl;
```

Comment:
Ok but why not using pugixml? I think it will be simpler to read and understand

Reply:
I just doesn't know how it works so it was quicker. I can have a look.